src/components/AliasMoreOptions.vue

```diff
@@ -0,0 +1,262 @@
+import { deepClone } from '../utils'
+import QuestionCircleIcon from '~icons/fa-solid/question-circle'
+import SaveIcon from '~icons/fa-solid/save'
+import TrashIcon from '~icons/fa-solid/trash'
+
+const props = withDefaults(
+  defineProps<{
+    alias: Alias
+    show: boolean
+    mailboxes: Mailbox[]
+    btnSaveLabel?: string
+  }>(),
+  {
+    btnSaveLabel: undefined
+  }
+)
+
+const emit = defineEmits<{
+  changed: [{ data: Alias }]
+  deleted: [{ data: Alias }]
+}>()
+
+const toast = useToast()
+const modal = useModalController()
+
+const moreOptions = ref<Alias>({
+  note: '',
+  name: '',
+  disable_pgp: false,
+  mailboxes: [],
+  email: '',
+  id: '',
+  support_pgp: true,
+  enabled: true,
+  nb_block: undefined,
+  nb_forward: undefined,
+  nb_reply: undefined,
+  showMoreOptions: false,
+  alias: '',
+  loading: false
+})
+const hasMailboxesChanges = ref(false) // to be used in canSaved()
+const canAlwaysSave = ref(false) // to be used in canSaved()
+
+onMounted(() => {
+  watch(
+    () => props.show,
+    (newValue) => {
+      if (newValue) {
+        showMoreOptions()
+      }
+    },
+    { immediate: true }
+  )
+
+  if (props.btnSaveLabel) {
+    canAlwaysSave.value = true
+  }
+})
+
+const showMoreOptions = () => {
+  moreOptions.value = {
+    note: props.alias.note,
+    name: props.alias.name,
+    disable_pgp: !!props.alias.disable_pgp,
+    mailboxes: deepClone(props.alias.mailboxes),
+    email: '',
+    id: '',
+    support_pgp: true,
+    enabled: true,
+    nb_block: undefined,
+    nb_forward: undefined,
+    nb_reply: undefined,
+    showMoreOptions: false,
+    alias: ''
+  }
+
+  hasMailboxesChanges.value = false
+}
+
+const handleClickDelete = async () => {
+  if (
+    await modal.confirm?.({
+      props: {
+        title: `Delete ${props.alias.email}`,
+        body: 'Do you really want to delete this alias?'
+      }
+    })
+  ) {
+    await deleteAlias()
+  }
+}
+
+const deleteAliasFetch = useDeleteAlias({
+  aliasId: () => props.alias.id,
+  onError: API_ON_ERR.TOAST
+})
+
+const deleteAlias = async () => {
+  await deleteAliasFetch.delete()
+  emit('deleted', {
+    data: props.alias
+  })
+}
+
+const canSave = () => {
+  return (
+    moreOptions.value.mailboxes.length > 0 &&
+    (props.alias.note !== moreOptions.value.note ||
+      props.alias.name !== moreOptions.value.name ||
+      !!props.alias.disable_pgp !== moreOptions.value.disable_pgp ||
+      hasMailboxesChanges.value ||
+      canAlwaysSave.value)
+  )
+}
+
+const editAliasFetch = usePutEditAlias({
+  aliasId: () => props.alias.id,
+  data: computed(() => ({
+    note: moreOptions.value.note,
+    name: moreOptions.value.name,
+    disable_pgp: moreOptions.value.disable_pgp,
+    mailbox_ids: moreOptions.value.mailboxes.map((mb) => mb.id)
+  }))
+})
+
+const loading = computed(() => deleteAliasFetch.isFetching.value || editAliasFetch.isFetching.value)
+
+const handleClickSave = async () => {
+  await editAliasFetch.put()
```

Comment:
Need to double check these api interactions. This is invalid syntax. SplashScreen.vue has a working one